TLSProfile tries to delete a referenced SSL Profile when making changes or deleting a VS

#### Setup Details
CIS Version : 2.12.0
Build: f5networks/k8s-bigip-ctlr:latest
BIGIP Version: Big IP 16.1.3.2
AS3 Version: 3.42.0
Agent Mode: AS3
Orchestration: K8S
Orchestration Version: 1.24.9
Pool Mode: Nodeport

#### Description  
 When assigning a ssl profile using TLSProfile crd, CIS tries to delete the reference profile when the TLSProfile changes or when the VS is removed. 

#### Steps To Reproduce
1) Crate a VS using a TLS profile, pointing the serverSSL or clientSSL to an existing bigip ssl profile
2) Delete the VS profile
3)The CIS will submit an AS# declaration to the Common partition to delete the profile that was reference.  

If TLSProfile does not create the ssl profile on the bigip, why does it try to delete it.

This is also the case modifying the TLSProfile.  

For example changing:
hosts:
   - "login.apps.test.local"
To
hosts:
   - "*.apps.cluster.local"   

Will also cause the CIS to push an AS3 declaration trying to delete the ssl profile that was referenced


#### Expected Result 
Since the the clientSSL and serverSSL are references to existing profiles, the profile should not be modified or deleted by CIS.

 apiVersion: cis.f5.com/v1
 kind: TLSProfile
 metadata:
   namespace: istio-system
   name: statusdemo-tls
   labels:
     f5cr: "true"
 spec:
   tls:
     termination: reencrypt
     serverSSL: /Common/istiogw-serverssl
     clientSSL: /Common/k8s_ingress_clientside_tls_profile
     reference: bigip            # --> reference profiles created in BIG-IP by User
   hosts:
   - "*.test.apps.local"
 
 CIS Log after a 'kubectl delete vs' posting to Common to delete the profile
2023/03/06 18:30:38 [ERROR] [AS3] Error response from BIG-IP: code: 422 --- tenant:**Common** --- message: declaration failed
2023/03/06 18:30:38 [DEBUG] [AS3] Posting failed tenants configuration in 30s seconds
   
BigIP Log
Mon, 06 Mar 2023 18:30:33 GMT - severe: [appsvcs] {"message":"Declaration failed: 01070294:3: The profile (/Common/Shared/k8s_ingress_clientside_tls_profile) is a parent of one or more profiles, can't delete.","level":"error"}

@lavanya-f5 Here is the full log of the following.
1) Created a TLS VS using pre-existing iRules and SSL profiles on /Common
2) changed the "allowVlans" on the VS just to cause a change
3) CIS will go into an endless 422 loop because it is unable to delete the iRules and SSL profiles in Common as thei are beeing used by the VS
4) I delete the VS "k delete vs" and CIS will delete the VS but also delete the referenced iRule and SSL Profiles in the /Common partition

2023/03/13 15:00:46 [DEBUG] [2023-03-13 15:00:46,732 __main__ DEBUG] config handler woken for reset
2023/03/13 15:00:46 [DEBUG] [2023-03-13 15:00:46,733 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:00:46 [DEBUG] [2023-03-13 15:00:46,734 __main__ DEBUG] NET Config: {}
2023/03/13 15:00:46 [DEBUG] [2023-03-13 15:00:46,734 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:00:46 [DEBUG] [2023-03-13 15:00:46,734 __main__ DEBUG] updating tasks finished, took 0.0018274784088134766 seconds
2023/03/13 15:01:16 [DEBUG] [2023-03-13 15:01:16,733 __main__ DEBUG] config handler woken for reset
2023/03/13 15:01:16 [DEBUG] [2023-03-13 15:01:16,733 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:01:16 [DEBUG] [2023-03-13 15:01:16,733 __main__ DEBUG] NET Config: {}
2023/03/13 15:01:16 [DEBUG] [2023-03-13 15:01:16,733 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:01:16 [DEBUG] [2023-03-13 15:01:16,733 __main__ DEBUG] updating tasks finished, took 0.00048661231994628906 seconds
2023/03/13 15:01:35 [INFO] Enqueueing TLSProfile: &{{ } {statusdemo-tls  istio-system  43d822bc-c321-4112-a02e-3f6b338d5a4e 9735324 1 2023-03-13 15:01:35 +0000 UTC <nil> <nil> map[f5cr:true] map[kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"cis.f5.com/v1","kind":"TLSProfile","metadata":{"annotations":{},"labels":{"f5cr":"true"},"name":"statusdemo-tls","namespace":"istio-system"},"spec":{"hosts":["statusdemo.pmilot.ecsdev.entrust.com"],"tls":{"clientSSL":"/Common/k8s_ingress_clientside_tls_profile","reference":"bigip","serverSSL":"/Common/istiogw-serverssl","termination":"reencrypt"}}}
] [] []  [{kubectl-client-side-apply Update cis.f5.com/v1 2023-03-13 15:01:35 +0000 UTC FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}},"f:labels":{".":{},"f:f5cr":{}}},"f:spec":{".":{},"f:hosts":{},"f:tls":{".":{},"f:clientSSL":{},"f:reference":{},"f:serverSSL":{},"f:termination":{}}}}}]} {[statusdemo.pmilot.ecsdev.entrust.com] {reencrypt /Common/k8s_ingress_clientside_tls_profile [] /Common/istiogw-serverssl [] bigip}}}
2023/03/13 15:01:35 [DEBUG] Processing Key: &{istio-system TLSProfile statusdemo-tls 0xc001e1c340 Create}
2023/03/13 15:01:35 [INFO] No VirtualServers found in namespace istio-system
2023/03/13 15:01:37 [DEBUG] Enqueueing VirtualServer: &{{ } {istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls  istio-system  58bdb1a0-d5e6-4aec-a98d-34df3821b919 9735337 1 2023-03-13 15:01:37 +0000 UTC <nil> <nil> map[f5cr:true] map[kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"cis.f5.com/v1","kind":"VirtualServer","metadata":{"annotations":{},"labels":{"f5cr":"true"},"name":"istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls","namespace":"istio-system"},"spec":{"host":"statusdemo.pmilot.ecsdev.entrust.com","iRules":["/Common/Shared/k8s_ingress_sni_irule"],"pools":[{"monitor":{"interval":3,"timeout":10,"type":"tcp"},"path":"/","service":"istio-ingressgateway","servicePort":443}],"snat":"auto","tlsProfileName":"statusdemo-tls","virtualServerAddress":"10.xx.xx.170","virtualServerName":"statusdemo-pmilot-ecsdev-entrust-com-tls"}}
] [] []  [{kubectl-client-side-apply Update cis.f5.com/v1 2023-03-13 15:01:37 +0000 UTC FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}},"f:labels":{".":{},"f:f5cr":{}}},"f:spec":{".":{},"f:host":{},"f:iRules":{},"f:pools":{},"f:snat":{},"f:tlsProfileName":{},"f:virtualServerAddress":{},"f:virtualServerName":{}}}}]} {statusdemo.pmilot.ecsdev.entrust.com  10.xx.xx.170 []  statusdemo-pmilot-ecsdev-entrust-com-tls 0 0 [{ / istio-ingressgateway {0 443 }  {tcp   3 10 0  } []     0  }] statusdemo-tls  auto   [] [/Common/Shared/k8s_ingress_sni_irule] []      {{ }      []  } [] false } { }}
2023/03/13 15:01:37 [DEBUG] Processing Key: &{istio-system VirtualServer istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls 0xc00081e700 Create}
2023/03/13 15:01:37 [DEBUG] Process all the Virtual Servers which share same VirtualServerAddress
2023/03/13 15:01:37 [DEBUG] Processing Virtual Server istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls for port 443
2023/03/13 15:01:37 [DEBUG] Configured rule: {vs_statusdemo_pmilot_ecsdev_entrust_com_istio_ingressgateway_443_istio_system_statusdemo_pmilot_ecsdev_entrust_com statusdemo.pmilot.ecsdev.entrust.com 0 [0xc000adeab0] [0xc001aee4b0]}
2023/03/13 15:01:37 [DEBUG] Configured policy: {statusdemo_pmilot_ecsdev_entrust_com_tls_443_statusdemo_pmilot_ecsdev_entrust_com_policy istio-system  [forwarding]  true [http] [0xc000ad4180] /Common/first-match}
2023/03/13 15:01:37 [DEBUG] Processing  BIGIP referenced profiles for 'VirtualServer' 'istio-system'/'istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls'
2023/03/13 15:01:37 [DEBUG] Updated BIGIP referenced profiles for 'VirtualServer' 'istio-system'/'istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls'
2023/03/13 15:01:37 [DEBUG] Updated Virtual istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls with TLSProfile statusdemo-tls
2023/03/13 15:01:37 [DEBUG] Finished syncing virtual servers &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls GenerateName: Namespace:istio-system SelfLink: UID:58bdb1a0-d5e6-4aec-a98d-34df3821b919 ResourceVersion:9735337 Generation:1 CreationTimestamp:2023-03-13 15:01:37 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[f5cr:true] Annotations:map[kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"cis.f5.com/v1","kind":"VirtualServer","metadata":{"annotations":{},"labels":{"f5cr":"true"},"name":"istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls","namespace":"istio-system"},"spec":{"host":"statusdemo.pmilot.ecsdev.entrust.com","iRules":["/Common/Shared/k8s_ingress_sni_irule"],"pools":[{"monitor":{"interval":3,"timeout":10,"type":"tcp"},"path":"/","service":"istio-ingressgateway","servicePort":443}],"snat":"auto","tlsProfileName":"statusdemo-tls","virtualServerAddress":"10.xx.xx.170","virtualServerName":"statusdemo-pmilot-ecsdev-entrust-com-tls"}}
] OwnerReferences:[] Finalizers:[] ClusterName: ManagedFields:[{Manager:kubectl-client-side-apply Operation:Update APIVersion:cis.f5.com/v1 Time:2023-03-13 15:01:37 +0000 UTC FieldsType:FieldsV1 FieldsV1:{"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}},"f:labels":{".":{},"f:f5cr":{}}},"f:spec":{".":{},"f:host":{},"f:iRules":{},"f:pools":{},"f:snat":{},"f:tlsProfileName":{},"f:virtualServerAddress":{},"f:virtualServerName":{}}}}]} Spec:{Host:statusdemo.pmilot.ecsdev.entrust.com HostGroup: VirtualServerAddress:10.xx.xx.170 AdditionalVirtualServerAddresses:[] IPAMLabel: VirtualServerName:statusdemo-pmilot-ecsdev-entrust-com-tls VirtualServerHTTPPort:0 VirtualServerHTTPSPort:0 Pools:[{Name: Path:/ Service:istio-ingressgateway ServicePort:{Type:0 IntVal:443 StrVal:} NodeMemberLabel: Monitor:{Type:tcp Send: Recv: Interval:3 Timeout:10 TargetPort:0 Name: Reference:} Monitors:[] Rewrite: Balance: WAF: ServiceNamespace: ReselectTries:0 ServiceDownAction: HostRewrite:}] TLSProfileName:statusdemo-tls HTTPTraffic: SNAT:auto WAF: RewriteAppRoot: AllowVLANs:[] IRules:[/Common/Shared/k8s_ingress_sni_irule] ServiceIPAddress:[] PolicyName: PersistenceProfile: ProfileMultiplex: DOS: BotDefense: Profiles:{TCP:{Client: Server:} UDP: HTTP: HTTP2: RewriteProfile: PersistenceProfile: LogProfiles:[] ProfileL4: ProfileMultiplex:} AllowSourceRange:[] HttpMrfRoutingEnabled:false Partition:} Status:{VSAddress: StatusOk:}} (189.96µs)
2023/03/13 15:01:37 [DEBUG] [CCCL] ConfigWriter (0xc0003c3440) writing section name gtm
2023/03/13 15:01:37 [DEBUG] [CCCL] ConfigWriter (0xc0003c3440) successfully wrote section (gtm)
2023/03/13 15:01:37 [DEBUG] Wrote gtm config section: map[]
2023/03/13 15:01:37 [DEBUG] [AS3] PostManager Accepted the configuration
2023/03/13 15:01:37 [DEBUG] [AS3] posting request to https://10.xx.xx.140/mgmt/shared/appsvcs/declare/pmrke2v1-cluster
2023/03/13 15:01:46 [DEBUG] [2023-03-13 15:01:46,733 __main__ DEBUG] config handler woken for reset
2023/03/13 15:01:46 [DEBUG] [2023-03-13 15:01:46,735 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:01:46 [DEBUG] [2023-03-13 15:01:46,736 __main__ DEBUG] NET Config: {}
2023/03/13 15:01:46 [DEBUG] [2023-03-13 15:01:46,736 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:01:46 [DEBUG] [2023-03-13 15:01:46,737 __main__ DEBUG] updating tasks finished, took 0.003373861312866211 seconds
2023/03/13 15:02:14 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:Common --- message: no change
2023/03/13 15:02:14 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:pmrke2v1-cluster --- message: success
2023/03/13 15:02:14 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:Common --- message: no change
2023/03/13 15:02:14 [DEBUG] Updating VirtualServer Status with { Ok} for resource name:istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls , namespace: istio-system
2023/03/13 15:02:14 [DEBUG] Enqueueing VirtualServer: &{{ } {istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls  istio-system  58bdb1a0-d5e6-4aec-a98d-34df3821b919 9735651 1 2023-03-13 15:01:37 +0000 UTC <nil> <nil> map[f5cr:true] map[kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"cis.f5.com/v1","kind":"VirtualServer","metadata":{"annotations":{},"labels":{"f5cr":"true"},"name":"istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls","namespace":"istio-system"},"spec":{"host":"statusdemo.pmilot.ecsdev.entrust.com","iRules":["/Common/Shared/k8s_ingress_sni_irule"],"pools":[{"monitor":{"interval":3,"timeout":10,"type":"tcp"},"path":"/","service":"istio-ingressgateway","servicePort":443}],"snat":"auto","tlsProfileName":"statusdemo-tls","virtualServerAddress":"10.xx.xx.170","virtualServerName":"statusdemo-pmilot-ecsdev-entrust-com-tls"}}
] [] []  [{kubectl-client-side-apply Update cis.f5.com/v1 2023-03-13 15:01:37 +0000 UTC FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}},"f:labels":{".":{},"f:f5cr":{}}},"f:spec":{".":{},"f:host":{},"f:iRules":{},"f:pools":{},"f:snat":{},"f:tlsProfileName":{},"f:virtualServerAddress":{},"f:virtualServerName":{}}}} {k8s-bigip-ctlr.real Update cis.f5.com/v1 2023-03-13 15:02:14 +0000 UTC FieldsV1 {"f:status":{".":{},"f:status":{},"f:vsAddress":{}}}}]} {statusdemo.pmilot.ecsdev.entrust.com  10.xx.xx.170 []  statusdemo-pmilot-ecsdev-entrust-com-tls 0 0 [{ / istio-ingressgateway {0 443 }  {tcp   3 10 0  } []     0  }] statusdemo-tls  auto   [] [/Common/Shared/k8s_ingress_sni_irule] []      {{ }      []  } [] false } {None Ok}}
2023/03/13 15:02:14 [DEBUG] Processing Key: &{istio-system VirtualServer istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls 0xc00079e000 Update}
2023/03/13 15:02:14 [DEBUG] Process all the Virtual Servers which share same VirtualServerAddress
2023/03/13 15:02:14 [DEBUG] Processing Virtual Server istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls for port 443
2023/03/13 15:02:14 [DEBUG] Configured rule: {vs_statusdemo_pmilot_ecsdev_entrust_com_istio_ingressgateway_443_istio_system_statusdemo_pmilot_ecsdev_entrust_com statusdemo.pmilot.ecsdev.entrust.com 0 [0xc0001b8990] [0xc0009fb0e0]}
2023/03/13 15:02:14 [DEBUG] Configured policy: {statusdemo_pmilot_ecsdev_entrust_com_tls_443_statusdemo_pmilot_ecsdev_entrust_com_policy istio-system  [forwarding]  true [http] [0xc0004649c0] /Common/first-match}
2023/03/13 15:02:14 [DEBUG] Processing  BIGIP referenced profiles for 'VirtualServer' 'istio-system'/'istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls'
2023/03/13 15:02:14 [DEBUG] Updated BIGIP referenced profiles for 'VirtualServer' 'istio-system'/'istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls'
2023/03/13 15:02:14 [DEBUG] Updated Virtual istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls with TLSProfile statusdemo-tls
2023/03/13 15:02:14 [DEBUG] Finished syncing virtual servers &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls GenerateName: Namespace:istio-system SelfLink: UID:58bdb1a0-d5e6-4aec-a98d-34df3821b919 ResourceVersion:9735651 Generation:1 CreationTimestamp:2023-03-13 15:01:37 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[f5cr:true] Annotations:map[kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"cis.f5.com/v1","kind":"VirtualServer","metadata":{"annotations":{},"labels":{"f5cr":"true"},"name":"istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls","namespace":"istio-system"},"spec":{"host":"statusdemo.pmilot.ecsdev.entrust.com","iRules":["/Common/Shared/k8s_ingress_sni_irule"],"pools":[{"monitor":{"interval":3,"timeout":10,"type":"tcp"},"path":"/","service":"istio-ingressgateway","servicePort":443}],"snat":"auto","tlsProfileName":"statusdemo-tls","virtualServerAddress":"10.xx.xx.170","virtualServerName":"statusdemo-pmilot-ecsdev-entrust-com-tls"}}
] OwnerReferences:[] Finalizers:[] ClusterName: ManagedFields:[{Manager:kubectl-client-side-apply Operation:Update APIVersion:cis.f5.com/v1 Time:2023-03-13 15:01:37 +0000 UTC FieldsType:FieldsV1 FieldsV1:{"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}},"f:labels":{".":{},"f:f5cr":{}}},"f:spec":{".":{},"f:host":{},"f:iRules":{},"f:pools":{},"f:snat":{},"f:tlsProfileName":{},"f:virtualServerAddress":{},"f:virtualServerName":{}}}} {Manager:k8s-bigip-ctlr.real Operation:Update APIVersion:cis.f5.com/v1 Time:2023-03-13 15:02:14 +0000 UTC FieldsType:FieldsV1 FieldsV1:{"f:status":{".":{},"f:status":{},"f:vsAddress":{}}}}]} Spec:{Host:statusdemo.pmilot.ecsdev.entrust.com HostGroup: VirtualServerAddress:10.xx.xx.170 AdditionalVirtualServerAddresses:[] IPAMLabel: VirtualServerName:statusdemo-pmilot-ecsdev-entrust-com-tls VirtualServerHTTPPort:0 VirtualServerHTTPSPort:0 Pools:[{Name: Path:/ Service:istio-ingressgateway ServicePort:{Type:0 IntVal:443 StrVal:} NodeMemberLabel: Monitor:{Type:tcp Send: Recv: Interval:3 Timeout:10 TargetPort:0 Name: Reference:} Monitors:[] Rewrite: Balance: WAF: ServiceNamespace: ReselectTries:0 ServiceDownAction: HostRewrite:}] TLSProfileName:statusdemo-tls HTTPTraffic: SNAT:auto WAF: RewriteAppRoot: AllowVLANs:[] IRules:[/Common/Shared/k8s_ingress_sni_irule] ServiceIPAddress:[] PolicyName: PersistenceProfile: ProfileMultiplex: DOS: BotDefense: Profiles:{TCP:{Client: Server:} UDP: HTTP: HTTP2: RewriteProfile: PersistenceProfile: LogProfiles:[] ProfileL4: ProfileMultiplex:} AllowSourceRange:[] HttpMrfRoutingEnabled:false Partition:} Status:{VSAddress:None StatusOk:Ok}} (180.298µs)
2023/03/13 15:02:16 [DEBUG] [2023-03-13 15:02:16,737 __main__ DEBUG] config handler woken for reset
2023/03/13 15:02:16 [DEBUG] [2023-03-13 15:02:16,738 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:02:16 [DEBUG] [2023-03-13 15:02:16,738 __main__ DEBUG] NET Config: {}
2023/03/13 15:02:16 [DEBUG] [2023-03-13 15:02:16,738 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:02:16 [DEBUG] [2023-03-13 15:02:16,738 __main__ DEBUG] updating tasks finished, took 0.0006890296936035156 seconds
2023/03/13 15:02:46 [DEBUG] [2023-03-13 15:02:46,735 __main__ DEBUG] config handler woken for reset
2023/03/13 15:02:46 [DEBUG] [2023-03-13 15:02:46,735 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:02:46 [DEBUG] [2023-03-13 15:02:46,735 __main__ DEBUG] NET Config: {}
2023/03/13 15:02:46 [DEBUG] [2023-03-13 15:02:46,735 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:02:46 [DEBUG] [2023-03-13 15:02:46,735 __main__ DEBUG] updating tasks finished, took 0.0004885196685791016 seconds
2023/03/13 15:03:16 [DEBUG] [2023-03-13 15:03:16,735 __main__ DEBUG] config handler woken for reset
2023/03/13 15:03:16 [DEBUG] [2023-03-13 15:03:16,736 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:03:16 [DEBUG] [2023-03-13 15:03:16,737 __main__ DEBUG] NET Config: {}
2023/03/13 15:03:16 [DEBUG] [2023-03-13 15:03:16,737 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:03:16 [DEBUG] [2023-03-13 15:03:16,737 __main__ DEBUG] updating tasks finished, took 0.0016734600067138672 seconds
2023/03/13 15:03:17 [DEBUG] Enqueueing VirtualServer: &{{ } {istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls  istio-system  58bdb1a0-d5e6-4aec-a98d-34df3821b919 9736147 2 2023-03-13 15:01:37 +0000 UTC <nil> <nil> map[f5cr:true] map[kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"cis.f5.com/v1","kind":"VirtualServer","metadata":{"annotations":{},"labels":{"f5cr":"true"},"name":"istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls","namespace":"istio-system"},"spec":{"allowVlans":["/Common/internal"],"host":"statusdemo.pmilot.ecsdev.entrust.com","iRules":["/Common/Shared/k8s_ingress_sni_irule"],"pools":[{"monitor":{"interval":3,"timeout":10,"type":"tcp"},"path":"/","service":"istio-ingressgateway","servicePort":443}],"snat":"auto","tlsProfileName":"statusdemo-tls","virtualServerAddress":"10.xx.xx.170","virtualServerName":"statusdemo-pmilot-ecsdev-entrust-com-tls"}}
] [] []  [{k8s-bigip-ctlr.real Update cis.f5.com/v1 2023-03-13 15:02:14 +0000 UTC FieldsV1 {"f:status":{".":{},"f:status":{},"f:vsAddress":{}}}} {kubectl-client-side-apply Update cis.f5.com/v1 2023-03-13 15:03:17 +0000 UTC FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}},"f:labels":{".":{},"f:f5cr":{}}},"f:spec":{".":{},"f:allowVlans":{},"f:host":{},"f:iRules":{},"f:pools":{},"f:snat":{},"f:tlsProfileName":{},"f:virtualServerAddress":{},"f:virtualServerName":{}}}}]} {statusdemo.pmilot.ecsdev.entrust.com  10.xx.xx.170 []  statusdemo-pmilot-ecsdev-entrust-com-tls 0 0 [{ / istio-ingressgateway {0 443 }  {tcp   3 10 0  } []     0  }] statusdemo-tls  auto   [/Common/internal] [/Common/Shared/k8s_ingress_sni_irule] []      {{ }      []  } [] false } {None Ok}}
2023/03/13 15:03:17 [DEBUG] Processing Key: &{istio-system VirtualServer istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls 0xc001aca380 Update}
2023/03/13 15:03:17 [DEBUG] Process all the Virtual Servers which share same VirtualServerAddress
2023/03/13 15:03:17 [DEBUG] Processing Virtual Server istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls for port 443
2023/03/13 15:03:17 [DEBUG] Configured rule: {vs_statusdemo_pmilot_ecsdev_entrust_com_istio_ingressgateway_443_istio_system_statusdemo_pmilot_ecsdev_entrust_com statusdemo.pmilot.ecsdev.entrust.com 0 [0xc000570120] [0xc001aef180]}
2023/03/13 15:03:17 [DEBUG] Configured policy: {statusdemo_pmilot_ecsdev_entrust_com_tls_443_statusdemo_pmilot_ecsdev_entrust_com_policy istio-system  [forwarding]  true [http] [0xc000ad5c80] /Common/first-match}
2023/03/13 15:03:17 [DEBUG] Processing  BIGIP referenced profiles for 'VirtualServer' 'istio-system'/'istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls'
2023/03/13 15:03:17 [DEBUG] Updated BIGIP referenced profiles for 'VirtualServer' 'istio-system'/'istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls'
2023/03/13 15:03:17 [DEBUG] Updated Virtual istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls with TLSProfile statusdemo-tls
2023/03/13 15:03:17 [DEBUG] Finished syncing virtual servers &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls GenerateName: Namespace:istio-system SelfLink: UID:58bdb1a0-d5e6-4aec-a98d-34df3821b919 ResourceVersion:9736147 Generation:2 CreationTimestamp:2023-03-13 15:01:37 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[f5cr:true] Annotations:map[kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"cis.f5.com/v1","kind":"VirtualServer","metadata":{"annotations":{},"labels":{"f5cr":"true"},"name":"istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls","namespace":"istio-system"},"spec":{"allowVlans":["/Common/internal"],"host":"statusdemo.pmilot.ecsdev.entrust.com","iRules":["/Common/Shared/k8s_ingress_sni_irule"],"pools":[{"monitor":{"interval":3,"timeout":10,"type":"tcp"},"path":"/","service":"istio-ingressgateway","servicePort":443}],"snat":"auto","tlsProfileName":"statusdemo-tls","virtualServerAddress":"10.xx.xx.170","virtualServerName":"statusdemo-pmilot-ecsdev-entrust-com-tls"}}
] OwnerReferences:[] Finalizers:[] ClusterName: ManagedFields:[{Manager:k8s-bigip-ctlr.real Operation:Update APIVersion:cis.f5.com/v1 Time:2023-03-13 15:02:14 +0000 UTC FieldsType:FieldsV1 FieldsV1:{"f:status":{".":{},"f:status":{},"f:vsAddress":{}}}} {Manager:kubectl-client-side-apply Operation:Update APIVersion:cis.f5.com/v1 Time:2023-03-13 15:03:17 +0000 UTC FieldsType:FieldsV1 FieldsV1:{"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}},"f:labels":{".":{},"f:f5cr":{}}},"f:spec":{".":{},"f:allowVlans":{},"f:host":{},"f:iRules":{},"f:pools":{},"f:snat":{},"f:tlsProfileName":{},"f:virtualServerAddress":{},"f:virtualServerName":{}}}}]} Spec:{Host:statusdemo.pmilot.ecsdev.entrust.com HostGroup: VirtualServerAddress:10.xx.xx.170 AdditionalVirtualServerAddresses:[] IPAMLabel: VirtualServerName:statusdemo-pmilot-ecsdev-entrust-com-tls VirtualServerHTTPPort:0 VirtualServerHTTPSPort:0 Pools:[{Name: Path:/ Service:istio-ingressgateway ServicePort:{Type:0 IntVal:443 StrVal:} NodeMemberLabel: Monitor:{Type:tcp Send: Recv: Interval:3 Timeout:10 TargetPort:0 Name: Reference:} Monitors:[] Rewrite: Balance: WAF: ServiceNamespace: ReselectTries:0 ServiceDownAction: HostRewrite:}] TLSProfileName:statusdemo-tls HTTPTraffic: SNAT:auto WAF: RewriteAppRoot: AllowVLANs:[/Common/internal] IRules:[/Common/Shared/k8s_ingress_sni_irule] ServiceIPAddress:[] PolicyName: PersistenceProfile: ProfileMultiplex: DOS: BotDefense: Profiles:{TCP:{Client: Server:} UDP: HTTP: HTTP2: RewriteProfile: PersistenceProfile: LogProfiles:[] ProfileL4: ProfileMultiplex:} AllowSourceRange:[] HttpMrfRoutingEnabled:false Partition:} Status:{VSAddress:None StatusOk:Ok}} (246.623µs)
2023/03/13 15:03:17 [DEBUG] [CCCL] ConfigWriter (0xc0003c3440) writing section name gtm
2023/03/13 15:03:17 [DEBUG] [CCCL] ConfigWriter (0xc0003c3440) successfully wrote section (gtm)
2023/03/13 15:03:17 [DEBUG] Wrote gtm config section: map[]
2023/03/13 15:03:17 [DEBUG] [AS3] PostManager Accepted the configuration
2023/03/13 15:03:17 [DEBUG] [AS3] posting request to https://10.xx.xx.140/mgmt/shared/appsvcs/declare/Common,pmrke2v1-cluster
2023/03/13 15:03:46 [DEBUG] [2023-03-13 15:03:46,735 __main__ DEBUG] config handler woken for reset
2023/03/13 15:03:46 [DEBUG] [2023-03-13 15:03:46,736 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:03:46 [DEBUG] [2023-03-13 15:03:46,736 __main__ DEBUG] NET Config: {}
2023/03/13 15:03:46 [DEBUG] [2023-03-13 15:03:46,736 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:03:46 [DEBUG] [2023-03-13 15:03:46,736 __main__ DEBUG] updating tasks finished, took 0.0006074905395507812 seconds
2023/03/13 15:03:54 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:Common --- message: success
2023/03/13 15:03:54 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:pmrke2v1-cluster --- message: success
2023/03/13 15:03:54 [ERROR] [AS3] Error response from BIG-IP: code: 422 --- tenant:Common --- message: declaration failed
2023/03/13 15:03:54 [DEBUG] [AS3] Posting failed tenants configuration in 30s seconds
2023/03/13 15:03:54 [DEBUG] Updating VirtualServer Status with {None Ok} for resource name:istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls , namespace: istio-system
2023/03/13 15:04:16 [DEBUG] [2023-03-13 15:04:16,736 __main__ DEBUG] config handler woken for reset
2023/03/13 15:04:16 [DEBUG] [2023-03-13 15:04:16,737 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:04:16 [DEBUG] [2023-03-13 15:04:16,737 __main__ DEBUG] NET Config: {}
2023/03/13 15:04:16 [DEBUG] [2023-03-13 15:04:16,737 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:04:16 [DEBUG] [2023-03-13 15:04:16,738 __main__ DEBUG] updating tasks finished, took 0.00093841552734375 seconds
2023/03/13 15:04:24 [DEBUG] [AS3] posting request to https://10.xx.xx.140/mgmt/shared/appsvcs/declare/Common
2023/03/13 15:04:46 [DEBUG] [2023-03-13 15:04:46,737 __main__ DEBUG] config handler woken for reset
2023/03/13 15:04:46 [DEBUG] [2023-03-13 15:04:46,737 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:04:46 [DEBUG] [2023-03-13 15:04:46,738 __main__ DEBUG] NET Config: {}
2023/03/13 15:04:46 [DEBUG] [2023-03-13 15:04:46,738 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:04:46 [DEBUG] [2023-03-13 15:04:46,738 __main__ DEBUG] updating tasks finished, took 0.0010440349578857422 seconds
2023/03/13 15:04:59 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:Common --- message: success
2023/03/13 15:04:59 [ERROR] [AS3] Error response from BIG-IP: code: 422 --- tenant:Common --- message: declaration failed
2023/03/13 15:04:59 [DEBUG] [AS3] Posting failed tenants configuration in 30s seconds
2023/03/13 15:04:59 [DEBUG] Updating VirtualServer Status with {None Ok} for resource name:istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls , namespace: istio-system
2023/03/13 15:05:16 [DEBUG] [2023-03-13 15:05:16,739 __main__ DEBUG] config handler woken for reset
2023/03/13 15:05:16 [DEBUG] [2023-03-13 15:05:16,740 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:05:16 [DEBUG] [2023-03-13 15:05:16,740 __main__ DEBUG] NET Config: {}
2023/03/13 15:05:16 [DEBUG] [2023-03-13 15:05:16,743 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:05:16 [DEBUG] [2023-03-13 15:05:16,744 __main__ DEBUG] updating tasks finished, took 0.004960298538208008 seconds
2023/03/13 15:05:29 [DEBUG] [AS3] posting request to https://10.xx.xx.140/mgmt/shared/appsvcs/declare/Common
2023/03/13 15:05:46 [DEBUG] [2023-03-13 15:05:46,739 __main__ DEBUG] config handler woken for reset
2023/03/13 15:05:46 [DEBUG] [2023-03-13 15:05:46,740 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:05:46 [DEBUG] [2023-03-13 15:05:46,740 __main__ DEBUG] NET Config: {}
2023/03/13 15:05:46 [DEBUG] [2023-03-13 15:05:46,742 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:05:46 [DEBUG] [2023-03-13 15:05:46,742 __main__ DEBUG] updating tasks finished, took 0.002182483673095703 seconds
2023/03/13 15:06:05 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:Common --- message: success
2023/03/13 15:06:05 [ERROR] [AS3] Error response from BIG-IP: code: 422 --- tenant:Common --- message: declaration failed
2023/03/13 15:06:05 [DEBUG] [AS3] Posting failed tenants configuration in 30s seconds
2023/03/13 15:06:05 [DEBUG] Updating VirtualServer Status with {None Ok} for resource name:istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls , namespace: istio-system
2023/03/13 15:06:12 [DEBUG] Enqueueing VirtualServer: &{{ } {istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls  istio-system  58bdb1a0-d5e6-4aec-a98d-34df3821b919 9737550 2 2023-03-13 15:01:37 +0000 UTC <nil> <nil> map[f5cr:true] map[kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"cis.f5.com/v1","kind":"VirtualServer","metadata":{"annotations":{},"labels":{"f5cr":"true"},"name":"istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls","namespace":"istio-system"},"spec":{"allowVlans":["/Common/internal"],"host":"statusdemo.pmilot.ecsdev.entrust.com","iRules":["/Common/Shared/k8s_ingress_sni_irule"],"pools":[{"monitor":{"interval":3,"timeout":10,"type":"tcp"},"path":"/","service":"istio-ingressgateway","servicePort":443}],"snat":"auto","tlsProfileName":"statusdemo-tls","virtualServerAddress":"10.xx.xx.170","virtualServerName":"statusdemo-pmilot-ecsdev-entrust-com-tls"}}
] [] []  [{k8s-bigip-ctlr.real Update cis.f5.com/v1 2023-03-13 15:02:14 +0000 UTC FieldsV1 {"f:status":{".":{},"f:status":{},"f:vsAddress":{}}}} {kubectl-client-side-apply Update cis.f5.com/v1 2023-03-13 15:03:17 +0000 UTC FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}},"f:labels":{".":{},"f:f5cr":{}}},"f:spec":{".":{},"f:allowVlans":{},"f:host":{},"f:iRules":{},"f:pools":{},"f:snat":{},"f:tlsProfileName":{},"f:virtualServerAddress":{},"f:virtualServerName":{}}}}]} {statusdemo.pmilot.ecsdev.entrust.com  10.xx.xx.170 []  statusdemo-pmilot-ecsdev-entrust-com-tls 0 0 [{ / istio-ingressgateway {0 443 }  {tcp   3 10 0  } []     0  }] statusdemo-tls  auto   [/Common/internal] [/Common/Shared/k8s_ingress_sni_irule] []      {{ }      []  } [] false } {None Ok}}
2023/03/13 15:06:12 [DEBUG] Processing Key: &{istio-system VirtualServer istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls 0xc00081e000 Delete}
2023/03/13 15:06:12 [DEBUG] Process all the Virtual Servers which share same VirtualServerAddress
2023/03/13 15:06:12 [DEBUG] Finished syncing virtual servers &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls GenerateName: Namespace:istio-system SelfLink: UID:58bdb1a0-d5e6-4aec-a98d-34df3821b919 ResourceVersion:9737550 Generation:2 CreationTimestamp:2023-03-13 15:01:37 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[f5cr:true] Annotations:map[kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"cis.f5.com/v1","kind":"VirtualServer","metadata":{"annotations":{},"labels":{"f5cr":"true"},"name":"istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls","namespace":"istio-system"},"spec":{"allowVlans":["/Common/internal"],"host":"statusdemo.pmilot.ecsdev.entrust.com","iRules":["/Common/Shared/k8s_ingress_sni_irule"],"pools":[{"monitor":{"interval":3,"timeout":10,"type":"tcp"},"path":"/","service":"istio-ingressgateway","servicePort":443}],"snat":"auto","tlsProfileName":"statusdemo-tls","virtualServerAddress":"10.xx.xx.170","virtualServerName":"statusdemo-pmilot-ecsdev-entrust-com-tls"}}
] OwnerReferences:[] Finalizers:[] ClusterName: ManagedFields:[{Manager:k8s-bigip-ctlr.real Operation:Update APIVersion:cis.f5.com/v1 Time:2023-03-13 15:02:14 +0000 UTC FieldsType:FieldsV1 FieldsV1:{"f:status":{".":{},"f:status":{},"f:vsAddress":{}}}} {Manager:kubectl-client-side-apply Operation:Update APIVersion:cis.f5.com/v1 Time:2023-03-13 15:03:17 +0000 UTC FieldsType:FieldsV1 FieldsV1:{"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}},"f:labels":{".":{},"f:f5cr":{}}},"f:spec":{".":{},"f:allowVlans":{},"f:host":{},"f:iRules":{},"f:pools":{},"f:snat":{},"f:tlsProfileName":{},"f:virtualServerAddress":{},"f:virtualServerName":{}}}}]} Spec:{Host:statusdemo.pmilot.ecsdev.entrust.com HostGroup: VirtualServerAddress:10.xx.xx.170 AdditionalVirtualServerAddresses:[] IPAMLabel: VirtualServerName:statusdemo-pmilot-ecsdev-entrust-com-tls VirtualServerHTTPPort:0 VirtualServerHTTPSPort:0 Pools:[{Name: Path:/ Service:istio-ingressgateway ServicePort:{Type:0 IntVal:443 StrVal:} NodeMemberLabel: Monitor:{Type:tcp Send: Recv: Interval:3 Timeout:10 TargetPort:0 Name: Reference:} Monitors:[] Rewrite: Balance: WAF: ServiceNamespace: ReselectTries:0 ServiceDownAction: HostRewrite:}] TLSProfileName:statusdemo-tls HTTPTraffic: SNAT:auto WAF: RewriteAppRoot: AllowVLANs:[/Common/internal] IRules:[/Common/Shared/k8s_ingress_sni_irule] ServiceIPAddress:[] PolicyName: PersistenceProfile: ProfileMultiplex: DOS: BotDefense: Profiles:{TCP:{Client: Server:} UDP: HTTP: HTTP2: RewriteProfile: PersistenceProfile: LogProfiles:[] ProfileL4: ProfileMultiplex:} AllowSourceRange:[] HttpMrfRoutingEnabled:false Partition:} Status:{VSAddress:None StatusOk:Ok}} (47.327µs)
2023/03/13 15:06:16 [DEBUG] [2023-03-13 15:06:16,740 __main__ DEBUG] config handler woken for reset
2023/03/13 15:06:16 [DEBUG] [2023-03-13 15:06:16,741 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:06:16 [DEBUG] [2023-03-13 15:06:16,741 __main__ DEBUG] NET Config: {}
2023/03/13 15:06:16 [DEBUG] [2023-03-13 15:06:16,741 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:06:16 [DEBUG] [2023-03-13 15:06:16,741 __main__ DEBUG] updating tasks finished, took 0.0007572174072265625 seconds
2023/03/13 15:06:35 [DEBUG] [AS3] posting request to https://10.xx.xx.140/mgmt/shared/appsvcs/declare/Common
2023/03/13 15:06:46 [DEBUG] [2023-03-13 15:06:46,741 __main__ DEBUG] config handler woken for reset
2023/03/13 15:06:46 [DEBUG] [2023-03-13 15:06:46,742 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:06:46 [DEBUG] [2023-03-13 15:06:46,743 __main__ DEBUG] NET Config: {}
2023/03/13 15:06:46 [DEBUG] [2023-03-13 15:06:46,743 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:06:46 [DEBUG] [2023-03-13 15:06:46,744 __main__ DEBUG] updating tasks finished, took 0.002512216567993164 seconds
2023/03/13 15:07:08 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:Common --- message: success
2023/03/13 15:07:08 [ERROR] [AS3] Error response from BIG-IP: code: 422 --- tenant:Common --- message: declaration failed
2023/03/13 15:07:08 [DEBUG] [AS3] Posting failed tenants configuration in 30s seconds
2023/03/13 15:07:08 [DEBUG] VirtualServer Not Found: istio-system/istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls
2023/03/13 15:07:16 [DEBUG] [2023-03-13 15:07:16,742 __main__ DEBUG] config handler woken for reset
2023/03/13 15:07:16 [DEBUG] [2023-03-13 15:07:16,742 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:07:16 [DEBUG] [2023-03-13 15:07:16,742 __main__ DEBUG] NET Config: {}
2023/03/13 15:07:16 [DEBUG] [2023-03-13 15:07:16,743 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:07:16 [DEBUG] [2023-03-13 15:07:16,743 __main__ DEBUG] updating tasks finished, took 0.0016834735870361328 seconds
2023/03/13 15:07:38 [DEBUG] [AS3] posting request to https://10.xx.xx.140/mgmt/shared/appsvcs/declare/Common
2023/03/13 15:07:46 [DEBUG] [2023-03-13 15:07:46,742 __main__ DEBUG] config handler woken for reset
2023/03/13 15:07:46 [DEBUG] [2023-03-13 15:07:46,743 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:07:46 [DEBUG] [2023-03-13 15:07:46,743 __main__ DEBUG] NET Config: {}
2023/03/13 15:07:46 [DEBUG] [2023-03-13 15:07:46,743 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:07:46 [DEBUG] [2023-03-13 15:07:46,746 __main__ DEBUG] updating tasks finished, took 0.0028357505798339844 seconds
2023/03/13 15:08:13 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:Common --- message: success
2023/03/13 15:08:13 [ERROR] [AS3] Error response from BIG-IP: code: 422 --- tenant:Common --- message: declaration failed
2023/03/13 15:08:13 [DEBUG] [CCCL] ConfigWriter (0xc0003c3440) writing section name gtm
2023/03/13 15:08:13 [DEBUG] VirtualServer Not Found: istio-system/istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls
2023/03/13 15:08:13 [DEBUG] [CCCL] ConfigWriter (0xc0003c3440) successfully wrote section (gtm)
2023/03/13 15:08:13 [DEBUG] Wrote gtm config section: map[]
2023/03/13 15:08:13 [DEBUG] [AS3] PostManager Accepted the configuration
2023/03/13 15:08:13 [DEBUG] [AS3] posting request to https://10.xx.xx.140/mgmt/shared/appsvcs/declare/Common,pmrke2v1-cluster
2023/03/13 15:08:16 [DEBUG] [2023-03-13 15:08:16,746 __main__ DEBUG] config handler woken for reset
2023/03/13 15:08:16 [DEBUG] [2023-03-13 15:08:16,747 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:08:16 [DEBUG] [2023-03-13 15:08:16,747 __main__ DEBUG] NET Config: {}
2023/03/13 15:08:16 [DEBUG] [2023-03-13 15:08:16,748 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:08:16 [DEBUG] [2023-03-13 15:08:16,748 __main__ DEBUG] updating tasks finished, took 0.0024220943450927734 seconds
2023/03/13 15:08:46 [DEBUG] [2023-03-13 15:08:46,747 __main__ DEBUG] config handler woken for reset
2023/03/13 15:08:46 [DEBUG] [2023-03-13 15:08:46,747 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:08:46 [DEBUG] [2023-03-13 15:08:46,748 __main__ DEBUG] NET Config: {}
2023/03/13 15:08:46 [DEBUG] [2023-03-13 15:08:46,748 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:08:46 [DEBUG] [2023-03-13 15:08:46,748 __main__ DEBUG] updating tasks finished, took 0.001222372055053711 seconds
2023/03/13 15:08:51 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:Common --- message: success
2023/03/13 15:08:51 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:pmrke2v1-cluster --- message: success
2023/03/13 15:08:51 [DEBUG] [AS3] Response from BIG-IP: code: 200 --- tenant:Common --- message: success
2023/03/13 15:08:51 [DEBUG] VirtualServer Not Found: istio-system/istio.combined.statusdemo.pmilot.ecsdev.entrust.com.tls
2023/03/13 15:09:16 [DEBUG] [2023-03-13 15:09:16,747 __main__ DEBUG] config handler woken for reset
2023/03/13 15:09:16 [DEBUG] [2023-03-13 15:09:16,747 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:09:16 [DEBUG] [2023-03-13 15:09:16,747 __main__ DEBUG] NET Config: {}
2023/03/13 15:09:16 [DEBUG] [2023-03-13 15:09:16,747 __main__ DEBUG] loaded configuration file successfully
2023/03/13 15:09:16 [DEBUG] [2023-03-13 15:09:16,749 __main__ DEBUG] updating tasks finished, took 0.0016088485717773438 seconds

--- a/docs/RELEASE-NOTES.rst
+++ b/docs/RELEASE-NOTES.rst
@@ -17,6 +17,9 @@ Added Functionality
 Bug Fixes
 ````````````
 * `Issue 2725 <https://github.com/F5Networks/k8s-bigip-ctlr/issues/2725>`_: AS3 label not working with AS3 configmap when filter-tenants set to true.
+* `Issue 2793 <https://github.com/F5Networks/k8s-bigip-ctlr/issues/2793>`_: TLSProfile crd not working when the SSL profile is from Shared location.
+* `Issue 2797 <https://github.com/F5Networks/k8s-bigip-ctlr/issues/2797>`_: TLSProfile deletes a referenced SSL Profile when making changes or deleting a VS.
+* `Issue 2799 <https://github.com/F5Networks/k8s-bigip-ctlr/issues/2799>`_: VirtualServer deletes a referenced iRule when making changes or deleting a VS.
 
 2.12.0
 -------------

--- a/pkg/controller/backend.go
+++ b/pkg/controller/backend.go
@@ -251,12 +251,16 @@ func (agent *Agent) agentWorker() {
 		agent.tenantResponseMap = make(map[string]tenantResponse)
 
 		for tenant := range agent.incomingTenantDeclMap {
-			if _, ok := agent.tenantPriorityMap[tenant]; ok {
-				priorityTenants = append(priorityTenants, tenant)
-			} else {
-				updatedTenants = append(updatedTenants, tenant)
+			// CIS with AS3 doesnt allow write to Common partition.So objects in common partition
+			// should not be updated or deleted by CIS. So removing from tenant map
+			if tenant != "Common" {
+				if _, ok := agent.tenantPriorityMap[tenant]; ok {
+					priorityTenants = append(priorityTenants, tenant)
+				} else {
+					updatedTenants = append(updatedTenants, tenant)
+				}
+				agent.tenantResponseMap[tenant] = tenantResponse{}
 			}
-			agent.tenantResponseMap[tenant] = tenantResponse{}
 		}
 
 		// Update the priority tenants first

--- a/pkg/controller/resourceConfig.go
+++ b/pkg/controller/resourceConfig.go
@@ -997,6 +997,13 @@ func ConvertStringToProfileRef(profileName, context, ns string) ProfileRef {
 	parts := strings.Split(profName, "/")
 	profRef := ProfileRef{Context: context, Namespace: ns, BigIPProfile: true}
 	switch len(parts) {
+	case 3:
+		// refernce to existing profile created using AS3 in Common(non-cis-managed) partition
+		if parts[1] == "Shared" {
+			profRef.Partition = parts[0] + "/" + parts[1]
+			profRef.Name = parts[2]
+		}
+
 	case 2:
 		profRef.Partition = parts[0]
 		profRef.Name = parts[1]

